render/detailed/connections.go

```diff
@@ -0,0 +1,197 @@
+				}
+				if isInternetNode(n) {
+					key.localAddr = localAddr
+				}
+				counts[key] = counts[key] + 1
+			}
+		}
+	}
+
+	columnHeaders := NormalColumns
+	if isInternetNode(n) {
+		columnHeaders = InternetColumns
+	}
+	return NodeSummaryGroup{
+		ID:         "incoming-connections",
+		TopologyID: topologyID,
+		Label:      "Inbound",
+		Columns:    columnHeaders,
+		Nodes:      connectionRows(counts, isInternetNode(n)),
+	}
+}
+
+func outgoingConnectionsTable(topologyID string, n render.RenderableNode, ns render.RenderableNodes) NodeSummaryGroup {
+	localEndpoints := endpointChildrenOf(n)
+
+	// For each node which has an edge FROM me
+	counts := map[connectionsRow]int{}
+	for _, node := range ns {
+		if !n.Adjacency.Contains(node.ID) {
+			continue
+		}
+		remoteNode := node.Copy()
+		remoteEndpointIDs := endpointChildIDsOf(remoteNode)
+
+		for _, localEndpoint := range localEndpoints {
+			_, localAddr, _, ok := render.ParseEndpointID(localEndpoint.ID)
+			if !ok {
+				continue
+			}
+
+			for _, remoteEndpointID := range localEndpoint.Adjacency.Intersection(remoteEndpointIDs) {
+				_, _, port, ok := render.ParseEndpointID(remoteEndpointID)
+				if !ok {
+					continue
+				}
+				key := connectionsRow{
+					localNode:  &n,
+					remoteNode: &remoteNode,
+					port:       port,
+				}
+				if isInternetNode(n) {
+					key.localAddr = localAddr
+				}
+				counts[key] = counts[key] + 1
+			}
+		}
+	}
+
+	columnHeaders := NormalColumns
+	if isInternetNode(n) {
+		columnHeaders = InternetColumns
+	}
+	return NodeSummaryGroup{
+		ID:         "outgoing-connections",
+		TopologyID: topologyID,
+		Label:      "Outbound",
+		Columns:    columnHeaders,
+		Nodes:      connectionRows(counts, isInternetNode(n)),
+	}
+}
+
+func endpointChildrenOf(n render.RenderableNode) []render.RenderableNode {
+	result := []render.RenderableNode{}
+	n.Children.ForEach(func(child render.RenderableNode) {
+		if _, _, _, ok := render.ParseEndpointID(child.ID); ok {
+			result = append(result, child)
+		}
+	})
+	return result
+}
+
+func endpointChildIDsOf(n render.RenderableNode) report.IDList {
+	result := report.MakeIDList()
+	n.Children.ForEach(func(child render.RenderableNode) {
+		if _, _, _, ok := render.ParseEndpointID(child.ID); ok {
+			result = append(result, child.ID)
+		}
+	})
+	return result
+}
+
+func isInternetNode(n render.RenderableNode) bool {
+	return n.ID == render.IncomingInternetID || n.ID == render.OutgoingInternetID
+}
+
+func connectionRows(in map[connectionsRow]int, includeLocal bool) []NodeSummary {
+	nodes := []NodeSummary{}
+	for row, count := range in {
+		id, label, linkable := row.remoteNode.ID, row.remoteNode.LabelMajor, true
+		if row.remoteAddr != "" {
+			id, label, linkable = row.remoteAddr+":"+row.port, row.remoteAddr, false
+		}
+		metadata := []MetadataRow{}
+		if includeLocal {
+			metadata = append(metadata,
+				MetadataRow{
+					ID:       "foo",
+					Value:    row.localAddr,
+					Datatype: number,
+				})
```

Comment:
> Because that is not what parseFloat does. Try it: `parseFloat("127.0.0.1")`

I did try it before my original comment. Try: `parseFloat("cloud.weave.works (52.3.116.14)")`.